test/corpus/instr_and_reg.test

```diff
@@ -51,6 +51,48 @@ andl	$15, %eax
         (int)
         (ident (reg))))
 
+==========
+Instruction -- Reg arg, Integer immediate arg (ARM A64)
+==========
+mov X10, #1
+---
+(program
+    (instruction
+        (word)
+        (ident
+            (reg
+                (word)))
+        (int)))
```

Comment:
This first test is related to this other change which was missing a specific test for it https://github.com/RubixDev/tree-sitter-asm/commit/f4032c9dbcc10dfbf9b302b2a449148483293593#diff-919ac210accac9ecc55a76d10a7590e3d85ca3f0e165b52d30f08faee486d0cb